File: client-scalaz/src/main/scala/org/openapitools/client/api/Category.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Category._

case class Category (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
name: Option[String],
shortName: Option[String],
description: Option[String],
`type`: Option[String],
application: Option[Application],
parent: Option[Category],
children: Option[List[Category]],
childrenCount: Option[Long],
display: Option[String],
sqootSlug: Option[String],
asset: Option[Asset],
externalId: Option[String],
externalType: Option[String],
favoriteCount: Option[Long],
root: Option[Category],
tree: Option[String],
parentName: Option[String],
parentId: Option[Long],
applicationId: Option[Long],
secondaryType: Option[String])

object Category {
  import DateTimeCodecs._

  implicit val CategoryCodecJson: CodecJson[Category] = CodecJson.derive[Category]
  implicit val CategoryDecoder: EntityDecoder[Category] = jsonOf[Category]
  implicit val CategoryEncoder: EntityEncoder[Category] = jsonEncoderOf[Category]
}

File: client-clojure/src/sirqul_io_t_platform/specs/category.clj
(ns sirqul-io-t-platform.specs.category
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.application :refer :all]
            [sirqul-io-t-platform.specs.category :refer :all]
            [sirqul-io-t-platform.specs.category :refer :all]
            [sirqul-io-t-platform.specs.asset :refer :all]
            [sirqul-io-t-platform.specs.category :refer :all]
            )
  (:import (java.io File)))


(def category-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :name) string?
   (ds/opt :shortName) string?
   (ds/opt :description) string?
   (ds/opt :type) string?
   (ds/opt :application) application-spec
   (ds/opt :parent) category-spec
   (ds/opt :children) (s/coll-of category-spec)
   (ds/opt :childrenCount) int?
   (ds/opt :display) string?
   (ds/opt :sqootSlug) string?
   (ds/opt :asset) asset-spec
   (ds/opt :externalId) string?
   (ds/opt :externalType) string?
   (ds/opt :favoriteCount) int?
   (ds/opt :root) category-spec
   (ds/opt :tree) string?
   (ds/opt :parentName) string?
   (ds/opt :parentId) int?
   (ds/opt :applicationId) int?
   (ds/opt :secondaryType) string?
   })

(def category-spec
  (ds/spec
    {:name ::category
     :spec category-data}))
